### DISCUSSION TITLE
Cannot remove 'files.associations' preferences

### DISCUSSION THREAD

**Discussion by saltylassi** [eclipse-theia/theia#12008] [General] [answered]
### Bug Description:

when remove ```files.associations``` for custom preferences editor widget, error occurs at ```FileSystemFrontendContribution.updateAssociations```

```Javascript 
- FileSystemFrontendContribution

    protected updateAssociations(): void {
        const fileAssociations = this.preferences['files.associations'];
        const mimeAssociations = Object.keys(fileAssociations).map(filepattern => ({ id: fileAssociations[filepattern], filepattern }));
        this.mimeService.setAssociations(mimeAssociations);
    }
```

```this.preferences['files.associations']``` becomes ```undefined```, and ```Object.keys(undefined)``` makes error.

here is my code:

```Javascript


function removePreferences(rebind: interfaces.Rebind, contrib: interfaces.ServiceIdentifier<unknown>, sch: PreferenceSchema): void {
    rebind(contrib).toConstantValue({
        schema: (() => {
            const s = sch;
            const schemaKeys = Object.keys(s.properties);
            schemaKeys.forEach(key => {
                delete s.properties[key];
            });

            return s;
        })()
    });

removePreferences(rebind, FileSystemPreferenceContribution, filesystemPreferenceSchema);
```

https://github.com/eclipse-theia/theia/commit/c5a7d712d21a7f89eb03c03244ac1b91f38d02a6#diff-f36fb1a0e913ea06a31998757f31c1f5a1841dd8ee33352257c9beb89771b818

I know file extension is necessary for checking language type, but that error annoys me. sorry

### Steps to Reproduce:

1. remove ```files.associations```
2. ```Uncaught TypeError TypeError: Cannot convert undefined or null to object```

### Additional Information

- Operating System: Windows10
- Theia Version: 1.29

**Comment by vince-fugnitto** [comment, marked as answer]
> I know file extension is necessary for checking language type, but that error annoys me. sorry

@saltylassi thank you for reporting the issue, I'm not entirely convinced it is a bug but if you're not interested in `files.associtations` (where you completely remove the preference) you can always override `updateAssociations` in your application.
